### PR TITLE
recipes-support: libfyaml for aktualizr-lite

### PR DESCRIPTION
**Pull request by ldts**
This tool (binary, not libraries) is used by aklite to convert yaml to json

Signed-off-by: Jorge Ramirez-Ortiz <jorge@foundries.io>

### CODE REVIEW

File: meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb [hunk 1]
```diff
@@ -0,0 +1,16 @@
+SUMMARY = "A fancy 1.2 YAML and JSON parser/writer."
+HOMEPAGE = "https://github.com/pantoniou/libfyaml"
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${S}/LICENSE;md5=6399094fbc639a289cfca2d660c010aa"
+
+FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}-${PV}:"
+
+SRC_URI = "git://github.com/pantoniou/libfyaml.git"
+SRCREV = "de89df9041b3a72b258680d6c8f6df31c5cee28a"
```
Comment (ricardosalveti) [inline meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb:9]: Do we need to use git or using the latest known release is enough? Released snapshots are preferred (then renaming the recipe from _git.bb to _0.6.bb).

File: meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+SUMMARY = "A fancy 1.2 YAML and JSON parser/writer."
+HOMEPAGE = "https://github.com/pantoniou/libfyaml"
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${S}/LICENSE;md5=6399094fbc639a289cfca2d660c010aa"
+
+FILESEXTRAPATHS_prepend := "${THISDIR}/${PN}-${PV}:"
```
Comment (ricardosalveti) [inline meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb:6]: No need for this line.

File: meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+SRC_URI = "git://github.com/pantoniou/libfyaml.git"
+SRCREV = "de89df9041b3a72b258680d6c8f6df31c5cee28a"
+
+DEPENDS = " pkgconf "
```
Comment (ricardosalveti) [inline meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb:11]: Is this needed?

File: meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+S = "${WORKDIR}/git"
+
+inherit autotools
+PARALLEL_MAKE = ""
```
Comment (ricardosalveti) [inline meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb:16]: Why setting parallel_make to ""?

File: meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb (hunk 3 above)
Comment (ldts) [reply meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb:11]: yes

File: meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb (hunk 4 above)
Comment (ldts) [reply meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb:16]: someone somewhere mentioned that parallel builds might mess autoconf (I didnt validate it though, but seemed possible)

File: meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb (hunk 2 above)
Comment (ldts) [reply meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb:6]: ok

File: meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb (hunk 1 above)
Comment (ldts) [reply meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb:9]: I havent tried, will test and update then

File: meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb (hunk 4 above)
Comment (ldts) [reply meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb:16]: will remove and check the build

**Comment by ldts** [comment]
@rsalveti these autotools dont seem to be installing in the final image?

jramirez@trex build-lmp$ find . -name "fy-tool"
./tmp-lmp/work/cortexa53-crypto-lmp-linux/libfyaml/0.6-r0/image/usr/bin/fy-tool
./tmp-lmp/work/cortexa53-crypto-lmp-linux/libfyaml/0.6-r0/package/usr/bin/.debug/fy-tool
./tmp-lmp/work/cortexa53-crypto-lmp-linux/libfyaml/0.6-r0/package/usr/bin/fy-tool
./tmp-lmp/work/cortexa53-crypto-lmp-linux/libfyaml/0.6-r0/packages-split/libfyaml/usr/bin/fy-tool
./tmp-lmp/work/cortexa53-crypto-lmp-linux/libfyaml/0.6-r0/packages-split/libfyaml-dbg/usr/bin/.debug/fy-tool
./tmp-lmp/work/cortexa53-crypto-lmp-linux/libfyaml/0.6-r0/build/src/fy-tool
./tmp-lmp/work/cortexa53-crypto-lmp-linux/libfyaml/0.6-r0/build/src/.libs/fy-tool


is this a bug or am I missing something in the recipe??

**Comment by ldts** [comment]
ok I'll cancel the PR until I figure out the install part. will try again later :)

**Comment by ldts** [comment]
added it to the factory.inc....although since it is only used by aktualizr-lite perhaps they should go together.
or with the ota-utils...dunno, so many choices...

**Comment by ricardosalveti** [comment]
> added it to the factory.inc....although since it is only used by aktualizr-lite perhaps they should go together.
> or with the ota-utils...dunno, so many choices...

If used/required by aktualizr-lite, just make it a dependency of aktualizr.

**Comment by ricardosalveti** [comment]
```
PACKAGECONFIG += "${@bb.utils.filter('SOTA_EXTRA_CLIENT_FEATURES', 'fiovb', d)}"
PACKAGECONFIG[fiovb] = ",,,optee-fiovb aktualizr-fiovb-env-rollback"
PACKAGECONFIG[ubootenv] = ",,u-boot-fw-utils,u-boot-fw-utils u-boot-default-env aktualizr-uboot-env-rollback"
```

You could extend that to cover libfyaml, like:
`PACKAGECONFIG[fyaml] = ",,,libfyaml"
`

And adding fyaml to PACKAGECONFIG by default.

**Comment by ldts** [comment]
right and I was thinking about messing with that but what would happen if other clients decide to use this tool? 
it is not much different than aktualizr-lite depending on docker-cli (actually is pretty much the same sort of dependency)
so I opted for just adding it to the factory inc.

"And adding fyaml to PACKAGECONFIG by default."

what do you mean? in lmp.inc? I am not too familiar with this bit...so is the typical use case of this construct for binaries calling other binaries? seems strange capturing dependencies between binaries in this way...

File: meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb (hunk 3 above)
Comment (ricardosalveti) [reply meta-lmp-base/recipes-support/libfyaml/libfyaml_git.bb]: yeah, that is preferred (inherit).

**Comment by ricardosalveti** [comment]
> right and I was thinking about messing with that but what would happen if other clients decide to use this tool?
> it is not much different than aktualizr-lite depending on docker-cli (actually is pretty much the same sort of dependency)
> so I opted for just adding it to the factory inc.
> 
> "And adding fyaml to PACKAGECONFIG by default."
> 
> ```
> PACKAGECONFIG += "libfyaml"
> PACKAGECONFIG[libfyaml] = ",,,libfyaml"
> ```

Right, this change would need to be part of the aktualizr_%.bbappend we have in meta-lmp-base.

> what do you mean something like the above? I am not too familiar with this bit...so is the typical use case of this construct for binaries calling other binaries? seems strange capturing dependencies between binaries in this way...

IIRC the syntax here is:

`PACKAGECONFIG[foobar] = "optiontoenableduringbuildtime,optiontodisableduringbuildtime,builddependency,runtimedependency"
`

So you could have this as a build option in aktualizr as well if needed. Since it seems you need during runtime, you have to add it at the last block.

File: meta-lmp-base/recipes-support/libfyaml/libfyaml_0.6.bb [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://${S}/LICENSE;md5=6399094fbc639a289cfca2d660c010aa"
+
+SRC_URI = "https://github.com/pantoniou/libfyaml/archive/refs/tags/v${PV}.tar.gz"
+SRC_URI[sha256sum] = "f28113cf3a3a62ccb6c73b0f24072cc8cee3af87531029718511419d38bc2621"
+
+S = "${WORKDIR}/libfyaml-${PV}"
+
+inherit autotools pkgconfig
+
```
Comment (ricardosalveti) [inline meta-lmp-base/recipes-support/libfyaml/libfyaml_0.6.bb:12]: Just drop this last empty line.

**Comment by ricardosalveti** [comment]
`WARNING: libfyaml-0.6-r0 do_package_qa: QA Issue: libfyaml: SRC_URI uses unstable GitHub archives [src-uri-bad]`

I think we need to use the release link instead.

**Comment by ricardosalveti** [comment]
https://github.com/pantoniou/libfyaml/archive/refs/tags/v0.6.tar.gz

> `WARNING: libfyaml-0.6-r0 do_package_qa: QA Issue: libfyaml: SRC_URI uses unstable GitHub archives [src-uri-bad]`
> 
> I think we need to use the release link instead.

Seems release is not available for 0.6, sigh.

In this case, use git then and use the hash from the 0.6 tag.

**Comment by ricardosalveti** [comment]
> right and I was thinking about messing with that but what would happen if other clients decide to use this tool?

Then they will need to rdepend on it as well.

In our case, as we know which tool is expecting to use it, it is better to just make it a dependency the tool we're already installing, as then we don't need to change the image.

**Comment by ldts** [comment]
the issue is that .tarball-version is missing in the tarball
should contain something like: 
`0.6.0`

File: meta-lmp-base/recipes-support/libfyaml/libfyaml_0.6.bb (hunk 5 above)
Comment (ldts) [reply meta-lmp-base/recipes-support/libfyaml/libfyaml_0.6.bb:12]: ok. do you know some magic to add a file to the tarball before the autotools recipe starts processing it?

**Comment by ldts** [comment]
I believe this should get merged rather than waiting for v.07 (or whatever version comes along). we can always bump it later

**Comment by ldts** [comment]
> I believe this should get merged rather than waiting for v.07 (or whatever version comes along). we can always bump it later

it doesnt matter anymore...just got an update from Pantelis

**Review by angolini — APPROVED** [review]

**Comment by mike-sul** [comment]
@ricardosalveti Can we merge it so I can test the latest aklite dev version (deps on fy-tool) against the latest lmp.

**Comment by ricardosalveti** [comment]
> @ricardosalveti Can we merge it so I can test the latest aklite dev version (deps on fy-tool) against the latest lmp.

Merging just the recipe itself is fine, as that won't influence anything for now, let me see if I can just cherry-pick that.

**Comment by ricardosalveti** [comment]
I pushed just the recipe at https://github.com/foundriesio/meta-lmp/commit/1dd1d1f487af829a7e477bd1b35e36ff693c3d11, @ldts please update this PR with just the aktualizr-lite addition once there is an updated rev that requires it.

File: meta-lmp-base/recipes-sota/aktualizr/aktualizr_%.bbappend [hunk 6]
```diff
@@ -19,6 +19,8 @@ SRC_URI_append_libc-musl = " \
 PACKAGECONFIG += "${@bb.utils.filter('SOTA_EXTRA_CLIENT_FEATURES', 'fiovb', d)}"
 PACKAGECONFIG[fiovb] = ",,,optee-fiovb aktualizr-fiovb-env-rollback"
 PACKAGECONFIG[ubootenv] = ",,u-boot-fw-utils,u-boot-fw-utils u-boot-default-env aktualizr-uboot-env-rollback"
+PACKAGECONFIG += "libfyaml"
```
[the remark this answers is not in the document]
Comment (ldts) [reply meta-lmp-base/recipes-sota/aktualizr/aktualizr_%.bbappend:22]: ok

[the remark this answers is not in the document]
Comment (ldts) [reply meta-lmp-base/recipes-sota/aktualizr/aktualizr_%.bbappend]: @ricardosalveti done

**Comment by ricardosalveti** [comment]
Manually merged after resolving a minor conflict and removing one patch which was already added in v80.

**Comment by mike-sul** [comment]
@ricardosalveti Thanks, so the libfyaml stuff is in the master/HEAD of lmp-manifest but is not in v80, which is the right thing to do.

**Pull request closed without merging**